jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Components.java

```diff
@@ -23,8 +23,11 @@
  */
 public interface Components
 {
+    // TODO remove?
     ByteBufferPool getByteBufferPool();
 
+    ByteBufferPool.Sized getSizedByteBufferPool();
+
```

Comment:
I think just have the one method that returns a Sized pool (but with javadoc saying what the sizes are).

```suggestion
    ByteBufferPool.Sized getByteBufferPool();
```

Reply:
I've changed my mind... we do need new methods because we have different sizes for input and output:

```suggestion
    ByteBufferPool getByteBufferPool();

    ByteBufferPool.Sized getInputByteBufferPool();
    ByteBufferPool.Sized getOutputByteBufferPool();
```

Reply:
See #12046 which includes an implementation of this and also shows that using it can greatly simplify code

Reply:
Agreed that this should be taken on in a different PR.